Source/Patches.cs

```diff
@@ -542,4 +1044,117 @@ public static void Postfix()
+					var freeColonistsCopy = map.mapPawns.FreeColonists.ToList();
+
+					foreach (var colonist in freeColonistsCopy)
+					{
+						foreach (var otherColonist in freeColonistsCopy)
+						{
+							if (colonist != otherColonist)
+							{
+								int currentOpinion = colonist.relations.OpinionOf(otherColonist);
+
+								if (lastOpinions.TryGetValue((colonist, otherColonist), out int previousOpinion))
+								{
+									string trend = currentOpinion == previousOpinion ? "stayed the same" :
+												   currentOpinion > previousOpinion ? "improved" : "worsened";
+
+									opinionMessages.Add($"{colonist.Name} current opinion of {otherColonist.Name}: {currentOpinion} (has {trend} since yesterday)");
+								}
+								else
+								{
+									opinionMessages.Add($"{colonist.Name} current opinion of {otherColonist.Name}: {currentOpinion}");
+								}
+
+								lastOpinions[(colonist, otherColonist)] = currentOpinion;
+							}
+						}
+					}
+
+					if (opinionMessages.Any())
+					{
+						string consolidatedMessage = string.Join("\n", opinionMessages);
+						Personas.Add(consolidatedMessage, 2);
+					}
+				}
+			}
+			catch (Exception e)
+			{
+				Log.Error($"Error in UIRoot_UIRootUpdate_OpinionReport_Patch: {e.Message}");
+			}
+		}
+	}
+	[HarmonyPatch(typeof(UIRoot), nameof(UIRoot.UIRootUpdate))]
+	public static class UIRoot_UIRootUpdate_ColonistThoughts_Patch
+	{
+		static int thoughtsTicks = -1;  // Initialize to -1 so that the first tick will increment to 0
+
+		public static void Postfix()
+		{
+			var map = Find.CurrentMap;
+			if (map == null || !map.IsPlayerHome) return;
+
+			thoughtsTicks++;
+			if (thoughtsTicks % 12000 == 0)
+			{
+
+				foreach (Pawn colonist in map.mapPawns.FreeColonists)
+				{
+					// A dictionary to hold the sum of mood effects for each thought type.
+					Dictionary<string, (int Count, float MoodEffect)> thoughtCounts = new Dictionary<string, (int, float)>();
+
+					foreach (Thought_Memory thought in colonist.needs.mood.thoughts.memories.Memories)
+					{
+						string label = thought.LabelCap;
+						float moodEffect = thought.MoodOffset();
+
+						if (thoughtCounts.ContainsKey(label))
+						{
+							thoughtCounts[label] = (thoughtCounts[label].Count + 1, thoughtCounts[label].MoodEffect + moodEffect);
+						}
+						else
+						{
+							thoughtCounts.Add(label, (1, moodEffect));
+						}
+					}
+
+					List<string> formattedThoughts = new List<string>
+						{
+							$"Thoughts of {colonist.Name.ToStringShort}:"
+						};
+
+					// Iterate over the dictionary to create the de-duplicated and summarized list of thoughts.
+					foreach (var kvp in thoughtCounts)
+					{
+						string plural = kvp.Value.Count > 1 ? "(x" + kvp.Value.Count + ")" : "";
+						formattedThoughts.Add($"{kvp.Key}{plural}: Mood Effect {kvp.Value.MoodEffect.ToStringWithSign()}");
+					}
+
+					string thoughtsMessage = string.Join("\n", formattedThoughts);
```

Comment:
Wouldn't that add a '\n' after the colon of "Thoughts of X:" and in general don't you want to keep information per context in a single line? (also better to debug imho). Not sure.

Reply:
This is how there thoughts look in the log:
> PRIO-2 Thoughts of Timofei:
Crashed together(x2): Mood Effect 0
Initial optimism: Mood Effect +10
Chitchat(x2): Mood Effect 0
PRIO-2 Thoughts of Cameron:
Crashed together(x2): Mood Effect 0
Initial optimism: Mood Effect +10
Chitchat(x2): Mood Effect 0
Deep talk: Mood Effect 0
PRIO-2 Thoughts of Tim:
Crashed together(x2): Mood Effect 0
Initial optimism: Mood Effect +10
Chitchat(x2): Mood Effect 0
Deep talk: Mood Effect 0
Ate without table: Mood Effect -3